File: 2022/03/rucksack.py
""" Advent of Code 2022. Day 3: Rucksack Reorganization """

with open("input.txt") as f:
    rucksacks = f.read().splitlines()

items = ""
for rucksack in rucksacks:
    r = len(rucksack) // 2
    items += set(rucksack[r:]).intersection(set(rucksack[:r])).pop()

priorities = " abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

print("Part 1:\t", sum(priorities.find(l) for l in items))

badges = [] 
for i in range(0, len(rucksacks), 3):
    first, *rest = [set(s) for s in rucksacks[i:i+3]]
    badges.append(first.intersection(*rest).pop())

print("Part 2:\t", sum(priorities.find(l) for l in badges))

File: 2022/03/rucksack.clj
(ns aoc.2022.03
  (:require
    [clojure.string :as str]
    [clojure.set :as set])
  )

(defn read-input []
  (-> "input.txt"
      (slurp)
      (str/split #"\n")
     ))

(def rucksacks (read-input))

(def lists (map #(partition (/ (count %) 2) %) rucksacks))
(def sets (for [l lists] (map set l)))

(def items (map #(apply set/intersection %) sets))
(def ascii (map int (map first items)))
(def priorities (map #(if (< % 96) (- % 38) (- % 96)) ascii))
(println (reduce + priorities))


